engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -163,6 +163,7 @@ public StramClient(Configuration conf, LogicalPlan dag) throws Exception
   {
     this.conf = conf;
     this.dag = dag;
+    dag.visitDAG();
```

Comment:
As of now the original use cases discussed in https://lists.apache.org/thread.html/1c37e0954cee029c9de537cc35ecb35beebdf49aba17de89bd2ce9ed@%3Cdev.apex.apache.org%3E does not require such functionality. But if needed could be added in easily in future.